src/configs/nuage/scripts/pps-simulator/simulate_pps_stats.py

```diff
@@ -333,128 +334,141 @@ def create_pre_defined_flows(self, **kwargs):
+                flow_record["SrcUplink"] = self.srcUplinks[srcUp]
+                flow_record["DstUplink"] = self.destUplinks[destUp]
+                flow_record["L7Classification"] = self.l7s[l7]
+                flow_record["AppID"] = appid
+                flow_record["Application"] = appname
+                flow_record["AppGroupID"] = appgrpid
+                flow_record["APMGroup"] = appgrpname
+                ingress_prob = get_random_with_prob(0.5)
+                if ingress_prob:
+                    flow_record["IngressBytes"] = inbytes
+                    flow_record["IngressMB"] = ingressMB
+                    flow_record["IngressPackets"] = inpkts
+                else:
+                    flow_record["IngressBytes"] = 0
+                    flow_record["IngressMB"] = 0
+                    flow_record["IngressPackets"] = 0
+
+                if not ingress_prob:
+                    flow_record["EgressBytes"] = inbytes
+                    flow_record["EgressMB"] = float(inbytes) / 1048576  # TODO: What is this number ?
+                    flow_record["EgressPackets"] = inpkts
+                else:
+                    flow_record["EgressBytes"] = 0
+                    flow_record["EgressMB"] = 0
+                    flow_record["EgressPackets"] = 0
+
+                flow_record["TotalBytesCount"] = inbytes + inbytes
+                flow_record["TotalMB"] = float(flow_record["TotalBytesCount"]) / 1048576
+
+                flow_record["TotalPacketsCount"] = inpkts + inpkts
+                flow_record["DstNSG"] = src_nsgid
+                flow_record["DestinationNSG"] = src_nsgname
+                flow_record["SlaStatus"] = self.slastatus[sla_status]
+                flow_record["HasSwitchedPaths"] = self.hasswitchedpath[sla_status]
+                flow_record["L7ClassEnhanced"] = self.l7s[l7]
+                flow_record["UnderlayID"] = underlayId
+                flow_record["UnderlayName"] = "Underlay"+str(underlayId)
+
+                json.dump(flow_record, flowstats, default=json_serial)
+                flowstats.write("\n")
+                index_name = "nuage_dpi_flowstats" + '_' + self.es_index_prefix
+                flow_record["_index"] = index_name
+                flow_record["_type"] = "nuage_doc_type"
+                self.flow_cnt += 1
+                yield flow_record
+            t_increment += 30
+
+    def generate_flow_stats(self, startTime, endTime,es_chunk_size):
         pre_flows = self.create_pre_defined_flows(appgrpids=self.appgrps, l7s=self.l7s,
                                          protos=self.protos,
                                          domains=self.domains, nsgs=self.nsgs,
                                          simData=self.simData)
-        sla_flows = []
 
-        flow_cnt = 0
-        sla_prob_new = 0.6
-        write_data = []
-        with open('/var/log/flowstats_new.log', 'w') as flowstats:
+        with open('log/flowstats_new.log', 'w') as flowstats:
```

Comment:
log folder is required .... missing in commit?

Reply:
Yes, should I add a folder creation command before this?
/var/log requires sudo permissions to write